### ISSUE TITLE
Not working on app engine

### ISSUE THREAD

**Issue by jsphpl** [piskvorky/smart_open#37]
When trying to use smart_open in App Engine, i get the following error:
```
from subprocess import _args_from_interpreter_flags
ImportError: cannot import name _args_from_interpreter_flags
```

Loosely translated through google, this reads as *there is no multiprocessing on app engine*, see
- [http://stackoverflow.com/questions/22235919/cannot-import-name-args-from-interpreter-flags](http://stackoverflow.com/questions/22235919/cannot-import-name-args-from-interpreter-flags)
- [https://groups.google.com/forum/#!topic/google-appengine/BCtU881-XgQ](https://groups.google.com/forum/#!topic/google-appengine/BCtU881-XgQ)

Any ideas how to tackle this? App Engine is very interesting platform for smart_open due to its lack of local disk access and limited memory.

**Comment by piskvorky** [comment]
Very interesting use case, thanks for reporting!

Yes, if `multiprocessing` breaks `smart_open` on App Engine, we should make it optional. Multiprocessing is only needed in a few places, it's optional and not critical in any way for smart_open to work.

Can you add some checks around this in a pull request, @jsphpl ?

I'm thinking some dynamic (runtime) check to see if multiprocessing is available (catch `ImportError`); if yes, proceed as normal. If not, do a `warnings.warn` to notify the user that he's running a gelded version of `smart_open`, but otherwise continue working.

**Comment by jsphpl** [comment]
Okay thank you, that sounds promising - i'll look at the code and see what i can do.

**Comment by jsphpl** [comment]
Fixed by #39. Tests still missing (see #47)
